pkg/controller/infrastructure/infraflow/ensure.go

```diff
@@ -91,16 +96,87 @@ func (fctx *FlowContext) ensureUserManagedVPC(ctx context.Context) error {
 	return nil
 }
 
-func (fctx *FlowContext) ensureSubnet(ctx context.Context) error {
-	var (
-		region = fctx.infra.Spec.Region
-	)
+func (fctx *FlowContext) ensureIPv6CIDRs(ctx context.Context) error {
+	nodeSubnet := fctx.whiteboard.GetObject(ObjectKeyNodeSubnet).(*compute.Subnetwork)
+	if nodeSubnet == nil {
+		return fmt.Errorf("failed to get the subnet for nodes")
+	}
+
+	nodesIPv6Range, err := fctx.computeClient.WaitForIPv6Cidr(ctx, fctx.infra.Spec.Region, fmt.Sprintf("%d", nodeSubnet.Id))
+	if err != nil {
+		return err
+	}
+	fctx.whiteboard.Set(NodesSubnetIPv6CIDR, nodesIPv6Range)
+
+	srvSubnet := fctx.whiteboard.GetObject(ObjectKeyServicesSubnet).(*compute.Subnetwork)
```

Comment:
Again consider using: `GetObject`
```suggestion
	srvSubnet := GetObject[*compute.Subnetwork](fctx.whiteboard, ObjectKeyServicesSubnet)
```